### DISCUSSION TITLE
Datetime with timezone in request doesn't work

### DISCUSSION THREAD

**Discussion by Pei116** [fastapi/fastapi#6973] [Questions] [answered]
```Python
from fastapi import FastAPI
from datetime import datetime
from ..models import Contact
from ..database import Database


app = FastAPI()

# Dependency
def get_db():
    db = Database()
    try:
        yield db
    finally:
        db.disconnect()

@app.get("/contacts/", response_model=List[Contact])
async def get_contacts(address: int, start_time: datetime, end_time: datetime, duration: int, distance: int, db: Database = Depends(get_db)):
    contacts = detect_contacts(db, address, start_time, end_time, duration, distance)
    return contacts
```

### Description

I want to get query parameters `start_time` and `end_time` as datetime values with timezone, based on ISO 8601 or RFC 3339.
It works without timezone, for example, "2021-01-19 16:00:00", but not with timezone, for example, "2021-01-19 16:00:00+05:00", returning such error:
```JSON
{
    "detail": [
        {
            "loc": [
                "query",
                "start_time"
            ],
            "msg": "invalid datetime format",
            "type": "value_error.datetime"
        }
    ]
}
```

### Environment

* OS: Docker container
* FastAPI Version: 0.63.0
* Python Version: 3.8.2

**Comment by Pei116** [comment]
FYI, it's explicitly mentioned in the documentation that it supports `ISO 8601` format for `datetime.datetime` type:
[Extra Data Type - FastAPI](https://fastapi.tiangolo.com/tutorial/extra-data-types/)

**Comment by falkben** [comment, marked as answer]
You don't show what your queries are, but I suspect that you might need to urlencode them.

Note that for ISO 8601, there shouldn't be a space in between the date and the time. There is a "T" in that character location. https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations

After removing your dependency, and some of the extra params, this was working for me:

```python
from datetime import datetime
from urllib.parse import urlencode

from fastapi import Body, FastAPI, Query, testclient

app = FastAPI()


@app.get("/contacts/")
async def get_contacts(
    start_time: datetime,
    address: int,
    distance: int,
):
    return start_time


client = testclient.TestClient(app)


def test_get_contacts():

    data = {
        "start_time": "2008-09-15T15:53:00+05:00",
        "address": 2,
        "distance": 3,
    }
    qs = urlencode(data)
    resp = client.get(f"/contacts/?{qs}")
    resp.raise_for_status()
```

w/ urlencoded query, you can even leave the space in, if you need it.

**Comment by Pei116** [comment]
@falkben Thank your for your hustle! Lemme double check and get back to you.

**Comment by raphaelauv** [comment]
@Pei116 can you close your issue , thank you

**Comment by tiangolo** [comment]
Thanks for the help here @falkben and @raphaelauv ! 👏  🙇 

Thanks for reporting back and closing the issue @Pei116  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
